import "RewardConfig.proto";

package OperaBattleConfig;

message StarReward
{
	required uint32 star = 1;  //星级
	required uint32 cash_cost	= 2; //消耗钻石
	required uint32 time_left	= 3;  //剩余时间
	repeated RewardConfig.Reward reward = 4; //奖励 
}

message DifficultyReward
{
	required uint32 index	= 1; //难度
	required uint32 time_limit	= 2; //难度时间
	required uint32 cost_food 	= 3; //消耗石油
	repeated StarReward star_reward	= 4;  //星级奖励
}

message BattleReward
{
	required uint32 id	= 1;  //剧本id
	repeated DifficultyReward difficulty = 2; //难度奖励
}

message BattleRewardAll
{
	repeated BattleReward battle_rewards	= 1;  //剧本奖励
}